napari/_vispy/layers/scalar_field.py

```diff
@@ -103,8 +103,8 @@ def _on_data_change(self) -> None:
             ndisplay, getattr(data, 'dtype', None)
         )
 
-        if ndisplay == 3 and self.layer.ndim == 2:
-            data = np.expand_dims(data, axis=0)
+        if ndisplay > data.ndim:
+            data = data.reshape((1,) * (ndisplay - data.ndim) + data.shape)
```

Comment:
Should this be in a different PR?

Reply:
Other than this change I think this is ready to merge. But I would like to keep independent fixes independent.